subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/BuildModelActionRunner.java

```diff
@@ -80,6 +76,9 @@ public void buildFinished(BuildResult result) {
         }
 
         private static BuildActionResult buildResult(GradleInternal gradle, BuildModelAction buildModelAction) {
+            if (!buildModelAction.isRunTasksOnly()) {
```

Comment:
❌ Same as above: `if (buildModelAction.isModelRequest()) { ... }` would make easier to understand the connection here.